tools/shared/src/main/scala/org/scalajs/core/tools/optimizer/OptimizerCore.scala

```diff
@@ -1332,17 +1368,19 @@ private[optimizer] abstract class OptimizerCore(
         true
 
       case _ =>
-        arg.tpe.base match {
-          case ClassType("s_Predef$$less$colon$less" | "s_Predef$$eq$colon$eq") =>
-            true
-          case _ =>
-            false
-        }
+        isTypeLikelyOptimizable(arg.tpe)
+    }
+
+    receiverAndArgs.exists(isLikelyOptimizable) || {
+      target.toString == "s_reflect_ClassTag$.apply__jl_Class__s_reflect_ClassTag" &&
+      (receiverAndArgs.tail.head match {
+        case PreTransTree(ClassOf(_), _) => true
+        case _                           => false
+      })
     }
-    receiverAndArgs.exists(isLikelyOptimizable)
   }
 
-  private def inline(allocationSite: Option[AllocationSite],
+  private def inline(allocationSites: Set[AllocationSite],
```

Comment:
Why is the `allocationSite` a `Set`? Wouldn't we want to consider the following two calls to be different:

```scala
val a = 1
val b = 2

foo(a, b)

def foo(x: Int, y: Int) = {
  if (x > y) foo(b, a)
  else x
}
```